model/simulationtests/daylighting_devices.rb

```diff
@@ -0,0 +1,151 @@
+
+windows = []
+spaces.each do |space|
+  surfaces = space.surfaces.sort_by { |s| s.name.to_s }
+  surfaces.each do |surface|
+    sub_surfaces = surface.subSurfaces.sort_by { |ss| ss.name.to_s }
+    sub_surfaces.each do |sub_surface|
+      next if sub_surface.subSurfaceType != 'FixedWindow'
+
+      windows << sub_surface
+    end
+  end
+end
+
+window1 = windows[0]
+window2 = windows[1]
+
+###############################################################################
+#                          DaylightingDeviceTubular                           #
+###############################################################################
+
+material = OpenStudio::Model::StandardOpaqueMaterial.new(model)
+material.setThickness(0.2032)
+material.setConductivity(1.3114056)
+material.setDensity(2242.8)
+material.setSpecificHeat(837.4)
+construction = OpenStudio::Model::Construction.new(model)
+construction.insertLayer(0, material)
+
+points = OpenStudio::Point3dVector.new
+points << OpenStudio::Point3d.new(0, 1, 0.2)
+points << OpenStudio::Point3d.new(0, 0, 0.2)
+points << OpenStudio::Point3d.new(1, 0, 0.2)
+points << OpenStudio::Point3d.new(1, 1, 0.2)
+dome = OpenStudio::Model::SubSurface.new(points, model)
+dome.setName('Dome')
+dome.setSubSurfaceType('TubularDaylightDome')
+dome.setSurface(window1.surface.get)
+
+points = OpenStudio::Point3dVector.new
+points << OpenStudio::Point3d.new(0, 1, 0.1)
+points << OpenStudio::Point3d.new(0, 0, 0.1)
+points << OpenStudio::Point3d.new(1, 0, 0.1)
+points << OpenStudio::Point3d.new(1, 1, 0.1)
+diffuser = OpenStudio::Model::SubSurface.new(points, model)
+diffuser.setName('Diffuser')
+diffuser.setSubSurfaceType('TubularDaylightDiffuser')
+diffuser.setSurface(window1.surface.get)
+
+tubular = OpenStudio::Model::DaylightingDeviceTubular.new(dome, diffuser, construction)
+tubular.setDiameter(1.1)
+tubular.setTotalLength(1.4)
+tubular.setEffectiveThermalResistance(0.28)
+
+transition_zone = OpenStudio::Model::TransitionZone.new(thermal_zone, 1)
+tubular.addTransitionZone(transition_zone)
+
+###############################################################################
+#                         DaylightingDeviceLightWell                          #
+###############################################################################
+
+sub_surface = window1
+
+light_well = OpenStudio::Model::DaylightingDeviceLightWell.new(sub_surface)
+light_well.setHeightofWell(1.2)
+light_well.setPerimeterofBottomofWell(12.0)
+light_well.setAreaofBottomofWell(9.0)
+light_well.setVisibleReflectanceofWellWalls(0.7)
+
+###############################################################################
+#                           DaylightingDeviceShelf                            #
+###############################################################################
+
+sub_surface = window2
+
+#### Inside shelf
+partitionGroup = OpenStudio::Model::InteriorPartitionSurfaceGroup.new(model)
+partitionGroup.setSpace(sub_surface.space.get)
```

Comment:
Inside shelf is an interior partition, so it must be on a partition group that has a space